src/main/scala/com/github/unchama/seichiassist/menus/nicknames/NicknameCombinationMenu.scala

```diff
@@ -0,0 +1,231 @@
+
+object NicknameCombinationMenu {
+
+  class Environment(
+    implicit val playerHeadSkinAPI: PlayerHeadSkinAPI[IO, Player],
+    implicit val ioCanOpenNicknameCombinationMenu: IO CanOpen NicknameCombinationMenu,
+    implicit val ioCanOpenNicknameMenu: IO CanOpen NickNameMenu.type
+  )
+
+  sealed trait NicknamePart {
+    def displayName: String
+  }
+
+  object NicknamePart {
+
+    case object Head extends NicknamePart {
+      override def displayName: String = "前"
+    }
+
+    case object Middle extends NicknamePart {
+      override def displayName: String = "中"
+    }
+
+    case object Tail extends NicknamePart {
+      override def displayName: String = "後"
+    }
+
+  }
+
+}
+
+case class NicknameCombinationMenu(pageIndex: Int = 0, nicknamePart: NicknamePart)
+    extends Menu {
+
+  import com.github.unchama.menuinventory.syntax._
+
+  private val MenuRowCount = 4
+
+  override type Environment = NicknameCombinationMenu.Environment
+
+  override val frame: MenuFrame =
+    MenuFrame(MenuRowCount.chestRows, s"$DARK_PURPLE${BOLD}二つ名組合せ「${nicknamePart.displayName}」")
+
+  private def liftedArchivementId: Seq[Int] = {
+    nicknamePart match {
+      case NicknamePart.Head => (1000 until 9900).filter(Nicknames.getNicknameFor(_).isDefined)
+      case NicknamePart.Middle =>
+        (9900 until 9999).filter(Nicknames.getNicknameFor(_).isDefined)
+      case NicknamePart.Tail => (1000 until 9900).filter(Nicknames.getNicknameFor(_).isDefined)
+    }
+  }
+
+  override def computeMenuLayout(
+    player: Player
+  )(implicit environment: Environment): IO[MenuSlotLayout] = IO {
+    val nicknameButtonPerPage = (MenuRowCount - 1).chestRows.slotCount
+
+    val nickNameCombinationMenuButtons = NickNameCombinationMenuButtons(player, nicknamePart)
+    val playerdata = SeichiAssist.playermap(player.getUniqueId)
+
+    import nickNameCombinationMenuButtons._
+
+    val archivementButtonMapping = liftedArchivementId
+      .slice(
+        nicknameButtonPerPage * pageIndex,
+        nicknameButtonPerPage * pageIndex + nicknameButtonPerPage
+      )
+      .zipWithIndex
+      .flatMap {
+        case (archivementId, index) =>
+          if (playerdata.TitleFlags.contains(archivementId)) {
+            val button = headPartButton(archivementId)
+
+            button.map(index -> _)
+          } else {
+            Some(index -> lockedNicknameButton(archivementId))
+          }
+      }
+
+    import eu.timepit.refined.auto._
+
+    val totalNumberOfPages = Math.ceil((liftedArchivementId.length + 1) / 27.0).toInt
+
+    val uiOperationButtonMapping = Map(
+      ChestSlotRef(3, 0) -> {
+        if (pageIndex != 0) Some(previousPageButton) else Some(toNicknameMenuButton)
+      },
+      ChestSlotRef(3, 4) -> Some(resetSelectionButton),
+      ChestSlotRef(3, 8) -> Option.when(pageIndex + 1 < totalNumberOfPages)(nextPageButton)
+    ).collect { case (slotRef, Some(button)) => slotRef -> button }
```

Comment:
やや認知負荷が高いので、`Option.when` を外側に追い出して `++` の右辺にするのはどうでしょうか:

```suggestion
    val uiOperationButtonMapping = Map(
      ChestSlotRef(3, 0) -> {
        if (pageIndex != 0) previousPageButton else toNicknameMenuButton
      },
      ChestSlotRef(3, 4) -> resetSelectionButton,
    ) ++ Option.when(pageIndex + 1 < totalNumberOfPages)(ChestSlotRef(3, 8) -> nextPageButton)
```

Reply:
`MapExtra.when` が定義されているので、それを使った形に書き換えました